tasks/libs/junit_upload_core.py

```diff
@@ -245,7 +245,15 @@ def produce_junit_tar(files, result_path):
         tgz.addfile(job_url_info, job_url_file)
 
 
-def repack_macos_junit_tar(infile, outfile):
+def repack_macos_junit_tar(workflow_conclusion, infile, outfile):
+    """
+    Repacks JUnit tgz file from macOS Github Action run, so it would
+    contain correct job name and job URL.
+    """
+    if workflow_conclusion == "cancelled" and not os.path.exists(infile):
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏
What about we just skip when the file we want to upload does not exist?

Reply:
I'm not sure what we are supposed to do if there is no file. I suppose we should have no file only when we run no tests (which can only happen in the fast test jobs I guess), whatever the status of the workflow. So except in case of cancelation I would expect a file each time, wdyt?